src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType: symbol = Symbol("UpdateStreamType");
+const CHECK_DELAY: number = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING: string = "dcosUIVersion";
+const DISMISSED_VERSION: string = "dismissedVersion";
+const LAST_TIME_CHECK: string = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+  [key: string]: DCOSUIVersion;
+}
+
+interface DCOSUIVersion {
+  dismissedVersion: string;
+  lastTimeCheck: number;
+  [key: string]: string | number;
+}
+
+export function showNotification(newVersion: string): void {
+  // TODO: Move inside UpdateServiceBanner
```

Comment:
❌ Isn't this part of this issue?

Reply:
Okay, gotcha, having the alert there and potentially merging it to master makes me worry though

Reply:
It used to be `console.log`, but typescript doesn't allow those.

Reply:
Typescript or tslint?

Reply:
You can do a `// tslint:disable-next-line`

Reply:
and therefore ignore the param being missing. or if it's unused just remove the function alltogether

Reply:
The function is useful for testing this PR and I think Mike also made use of it.